spec/requests/users_controller_spec.rb

```diff
@@ -3071,6 +3071,54 @@ def post_user(extra_params = {})
         expect(response.status).to eq(200)
       end
     end
+
+    context 'avatar flair in Most... sections' do
+      it "returns data for automatic groups flair" do
+        liker = Fabricate(:user, admin: true, moderator: true, trust_level: 1)
```

Comment:
Why does this user needs to be both admin and mod?

Reply:
Code that decides how to display here is on the client-side, here [user-avatar-flair](https://github.com/discourse/discourse/blob/master/app/assets/javascripts/discourse/app/components/user-avatar-flair.js) and [here](https://github.com/discourse/discourse/blob/cd93d1b5f76aa8b696783f2c5a7edf2920ef8254/app/assets/javascripts/discourse/app/lib/avatar-flair.js#L3).

All that the server should do for correct displaying of the flair of an automatic group is to return to the client-side:
- user.admin
- user.moderator
- user.trust_level

With this information, the client is able to decide how to display flair. If we forget to return one of this field flair may be shown incorrectly.

Reply:
Right, since this is mostly information that is shown on the UI, wouldn't it be better to have it tested in an acceptance test rather than a unit test?

Reply:
But in acceptance tests, we use a pretender. We would return proper JSON and it would pass. But the problem that this PR and the previous one are fixing is the lack of information that the server returns. 

An acceptance test wouldn't catch this bug.

I can add an acceptance test though, but anyway we should decide if we want to leave these pretty weird server tests in the code base. I agree that they are pretty weird, that's why I didn't add tests in the beginning.

I would leave them, cause they do the job and document current implementation.

Then later we can move [this](https://github.com/discourse/discourse/blob/cd93d1b5f76aa8b696783f2c5a7edf2920ef8254/app/assets/javascripts/discourse/app/lib/avatar-flair.js#L3) logic to the server.

Reply:
Good point 👍